cmd/serve.go

```diff
@@ -96,16 +103,29 @@ func serve(_ context.Context) {
 		logger.Fatal("error initializing API server: %s", err)
 	}
 
-	userInfoHandler, err := userinfo.NewHandler(storageEngine, oauth2Config)
+	userInfoHandler, err := userinfo.NewHandler(storageEngine)
 	if err != nil {
 		logger.Fatal("error initializing UserInfo handler: %s", err)
 	}
 
 	router := routes.NewRouter(logger, oauth2Config, provider, config.Config.OAuth.Issuer)
 
+	authMdwSkippers := []middleware.Skipper{
+		echox.SkipDefaultEndpoints,
+		routes.SkipNoAuthRoutes,
+	}
+
+	var middleware []echo.MiddlewareFunc
+
+	if authMdw, err := getAuthMiddleware(ctx, oauth2Config, authMdwSkippers...); err != nil {
+		logger.Fatal("failed to initialize jwt authentication", zap.Error(err))
+	} else if authMdw != nil {
+		middleware = append(middleware, authMdw)
+	}
```

Comment:
Are there conditions under which `auth.Middleware()` might return `nil`? The docs don't make it 100% clear, but from this code it looks like it could happen: https://pkg.go.dev/go.infratographer.com/x/echojwtx#Auth.Middleware

Reply:
That is a good point. I think that might've been left over from a previous bit where auth could be disabled. Which is not able to be done in this case. fixed!